meshery/meshery
# [mesheryctl] platform: kubernetes support for `system status` command

Signed-off-by: navendu-pottekkat <navendupottekkat@gmail.com>

**Description**

This PR fixes #2514

Running `mesheryctl system status` when the `platform` is set to `kubernetes` (in the `config.yaml` file) will list all the ~~pods~~ **deployments** running in the `MesheryNamespace`(currently `meshery`)-

![Screenshot from 2021-03-16 13-54-19](https://user-images.githubusercontent.com/49474499/111280354-b9acee80-8661-11eb-97b8-b03ba6caf7fe.png)

If the `platform` is `docker`, nothing is changed and the Meshery containers are shown.

**Notes for Reviewers**

**Also changed every occurrence of using a local function to create a** `kube client`. **Instead, changed to** `New` **function from MeshKit.**

To Test this PR-

1. Run Meshery on a Kubernetes cluster. (Change `platform` to `kubernetes` in the `config.yaml` file and run `mesheryctl system start`)
2. Run `mesheryctl system status` (Which should have the same output as `kubectl get deployments -n meshery`)

**Future**

Move the `PrintToTable` function to MeshKit if there are more uses for it.
